src/ejabberd_c2s.erl

```diff
@@ -286,7 +285,7 @@ stream_start_error(Error, StateData) ->
       State :: state(),
       Result :: {stop, normal, state()}.
 c2s_stream_error(Error, StateData) ->
-    send_element(StateData, Error),
+    send_element_from_server(StateData, Error),
```

Comment:
I know naming can be difficult, and there will always be someone not happy with a name for a function :) In this particular case I think better name would be `send_element_to_client` because we are in the c2s process and in most cases we want to sent the element to the client represented by the c2s process. In my opinion it's obvious we are sending from the server, it's not that obvious where we are sending. Someone else may think in the other way around though :)

Reply:
I think `send_element_from_server_domain` would be more accurate, because this function is meant specifically for stanzas sent by server logic, not from another user.